### ISSUE TITLE
Scroll event keeps firing

### ISSUE THREAD

**Issue by chrishawn** [orizens/angular2-infinite-scroll#7]
Does not seem to respect infiniteScrollDistance setting.
Is infiniteScrollDistance implented?

i have 
```
<div #itemListViewWrapper infinite-scroll [infiniteScrollDistance]="1" (scrolled)="onScroll()">
    <item-component *ngFor="#item of itemListView" [item]="item"></item-component>
</div>
```

onScroll fires basically whenever i scroll regardless of distance bottom of the div to bottom of window
changing it to 0 or 2 doesn't matter

**Comment by chrishawn** [comment]
what i have done so far to get it working, this does not jive with how the angular1 version worked

```
    onScroll () {
        console.log('scrolled!!');
        var items=this.articles.nativeElement.getElementsByTagName('item');
        var lastEl=items[items.length-1];
        if(lastEl && this.isScrolledIntoView(lastEl)){ //if last article in view then load next article
            if(this.itemList.length>=arts.length) {
                this.itemList.push(this.itemList[items.length]);
                console.log('adding item')
            }
        }
    }

    isScrolledIntoView(el) {
        return (window.scrollY+window.innerHeight)>el.offsetTop;
    }
```

**Comment by orizens** [comment]
hi @chrishawn 
if you can integrate that as a pr, i'de be happy to merge it.

**Comment by zoehneto** [comment]
I had the same issue and, at least in my case, it was due `elem.document.documentElement.clientHeight` not returning the viewport height but the document height in the `Scroller.height()` function, which led to every scroll event triggering the callback. After replacing it with `elem.document.body.clientHeight` it works for me in Chrome 50.

**Comment by zoehneto** [comment]
It turns out that in my case the issue was actually caused by a missing `DOCTYPE` declaration which leads to the output of `documentElement.clientHeight`and `body.clientHeight` being switched for compatibility with older websites as described [here](http://javascript.info/draft/doctype-strict-almost-strict-and-quirks-modes).

**Comment by orizens** [comment]
this should be fixed in the next release soon.

**Comment by orizens** [comment]
this issue has been addressed with this release - f5534b5ec09f0517725b4a63bfd55fb80bcc41ca

**Comment by justme1** [comment]
@zoehneto  amazing catch regarding missing `DOCTYPE` - I experienced the same issue!

@orizens , I suggest this one being updated and reflected in the README.
